src/main/java/org/opentripplanner/middleware/controllers/api/MonitoredTripController.java

```diff
@@ -0,0 +1,59 @@
+package org.opentripplanner.middleware.controllers.api;
+
+import com.mongodb.client.model.Filters;
+import org.bson.conversions.Bson;
+import org.eclipse.jetty.http.HttpStatus;
+import org.opentripplanner.middleware.models.MonitoredTrip;
+import org.opentripplanner.middleware.persistence.Persistence;
+import spark.Request;
+
+import static com.mongodb.client.model.Filters.eq;
+import static org.opentripplanner.middleware.auth.Auth0Connection.isAuthorized;
+import static org.opentripplanner.middleware.spark.Main.getConfigPropertyAsInt;
+import static org.opentripplanner.middleware.utils.JsonUtils.logMessageAndHalt;
+
+/**
+ * Implementation of the {@link ApiController} abstract class for managing monitored trips. This controller connects
+ * with Auth0 services using the hooks provided by {@link ApiController}.
+ */
+public class MonitoredTripController extends ApiController<MonitoredTrip> {
+    private static final int MAXIMUM_PERMITTED_MONITORED_TRIPS
+        = getConfigPropertyAsInt("MAXIMUM_PERMITTED_MONITORED_TRIPS", 5);
+
+    public MonitoredTripController(String apiPrefix) {
+        super(apiPrefix, Persistence.monitoredTrip, "secure/monitoredtrip");
+    }
+
+    @Override
+    MonitoredTrip preCreateHook(MonitoredTrip monitoredTrip, Request req) {
+        isAuthorized(monitoredTrip.userId, req);
+        verifyBelowMaxNumTrips(monitoredTrip.userId, req);
+
+        return monitoredTrip;
+    }
+
+    @Override
+    MonitoredTrip preUpdateHook(MonitoredTrip monitoredTrip, MonitoredTrip preExisting, Request req) {
+        isAuthorized(monitoredTrip.userId, req);
+        return monitoredTrip;
+    }
+
+    @Override
+    boolean preDeleteHook(MonitoredTrip monitoredTrip, Request req) {
+        // Authorization checks are done prior to this hook
```

Comment:
It seems a little off that we would handle authorization differently for delete vs. update/create. Also, I noticed a potential bug in the isAuthorized check. I think it's possible that Auth0UserProfile could have both an apiUser and an otpUser. The way the check works now, I think it will throw a halt if (e.g.) the otpUser.id matches the monitored trip userId, but the apiUser.id does not.

Reply:
We would need to decide if the delete auth is to match the update/create auth... as all controllers will need to be updated to account for this.

I have separated the otpUser and apiUser userid checks.